llama_stack/apis/common/job_types.py

```diff
@@ -3,21 +3,42 @@
 #
 # This source code is licensed under the terms described in the LICENSE file in
 # the root directory of this source tree.
+from datetime import datetime
 from enum import Enum
+from typing import Optional
 
 from pydantic import BaseModel
 
 from llama_stack.schema_utils import json_schema_type
 
 
-@json_schema_type
-class Job(BaseModel):
-    job_id: str
-
-
-@json_schema_type
 class JobStatus(Enum):
     completed = "completed"
     in_progress = "in_progress"
     failed = "failed"
     scheduled = "scheduled"
+    cancelled = "cancelled"
+
+
+class JobType(Enum):
+    batch_inference = "batch_inference"
+    scoring = "scoring"
+    evaluation = "evaluation"
+    post_training = "post_training"
+
+
+@json_schema_type
+class CommonJobFields(BaseModel):
+    """Common fields for all jobs.
+    :param id: The ID of the job.
+    :param status: The status of the job.
+    :param created_at: The time the job was created.
+    :param finished_at: The time the job finished.
+    :param error: If status of the job is failed, this will contain the error message.
+    """
+
+    id: str
+    status: JobStatus
+    created_at: datetime
+    finished_at: Optional[datetime] = None
+    error: Optional[str] = None
```

Comment:
Should it be a single error, or multiple errors (list)?

---

I was thinking of common traits of jobs, and one idea I had is that all of them would probably have "logs". How would I model it with the proposed API? Should there be a separate (streaming?) API for log events? Or is it an "artifact"? (result?) (If so, how one would stream logs?)

Reply:
> (If so, how one would stream logs?)

Would a separate `/events` & `/logs` make sense here?

Reply:
They may! (btw it may make sense to leave these capabilities out of the picture for this PR; I just want to make sure we have some understanding of how this can be modeled in the future.)

---

So about `error` field: do you think we'll need it as such if we have `/logs`?